### ISSUE TITLE
Misleading error on 'convox uninstall' ("ERROR: Can not find rack named <name>") when credentials are needed

### ISSUE THREAD

**Issue by soulshake** [convox/rack#1416]
# TL;DR

When I try to uninstall a Rack without the needed credentials, I get a misleading error (`ERROR: Can not find rack named python`).

## What I do

Verify that I have a Rack `personal/python`:

```
$ convox rack
Name     python
Status   running
Version  20161102160040
Region   eu-central-1
Count    5
Type     t2.small
```

```
$ convox switch
personal/python
```

... and try to uninstall it by running `convox uninstall`.

## What I expect

I expect any of the following:

- for the Rack to be uninstalled,
- to receive an error message because I haven't provided credentials, OR
- to be prompted for credentials.

## What I get instead

```
$ convox uninstall python eu-central-1
     ___    ___     ___   __  __    ___   __  _
    / ___\ / __ \ /  _  \/\ \/\ \  / __ \/\ \/ \
   /\ \__//\ \_\ \/\ \/\ \ \ \_/ |/\ \_\ \/>  </
   \ \____\ \____/\ \_\ \_\ \___/ \ \____//\_/\_\
    \/____/\/___/  \/_/\/_/\/__/   \/___/ \//\/_/

ERROR: Can not find rack named python.
```

## Workaround

It worked when I specified the credentials file:

```
$ convox uninstall python eu-central-1 ~/.secrets/convox.guest.csv


     ___    ___     ___   __  __    ___   __  _
    / ___\ / __ \ /  _  \/\ \/\ \  / __ \/\ \/ \
   /\ \__//\ \_\ \/\ \/\ \ \ \_/ |/\ \_\ \/>  </
   \ \____\ \____/\ \_\ \_\ \___/ \ \____//\_/\_\
    \/____/\/___/  \/_/\/_/\/__/   \/___/ \//\/_/


Resources to delete:

STACK           TYPE     STATUS
redis-6718      service  CREATE_COMPLETE
convox-events   service  CREATE_COMPLETE
trainingwheels  app      UPDATE_COMPLETE
python          rack     UPDATE_COMPLETE

Delete everything? y/N: y

Uninstalling Convox...
Deleting redis-6718...
Deleting convox-events...
Uninstalling Convox...
[...]
```

**Comment by MiguelMoll** [comment]
This could be tricky as a set of credentials might not be able to see a Rack while another set can. Be it for lack of permissions or its another AWS account entirely. 

Without providing a credentials file, the cli uses standard AWSCLI creds from [environment variables](http://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html#cli-environment) or `~/.aws/credentials` file. Could you verify if those creds have the same access as `~/.secrets/convox.guest.csv`?

**Comment by soulshake** [comment]
@MiguelMoll I don't have `~/.aws/credentials` (I don't remember if that file existed at the time I created this issue).

Assuming it did exist, I'm sure the two credential files didn't have the same access, which helps me see where it's coming from: `Can not find rack named python` == "the credentials we found in the default location don't have access to a CloudFormation stack named `python`."

I still think the error is confusing, but I think now I have enough info to be able to reword it to something more clear.

Maybe instead of:

`ERROR: Can not find rack named python.`

we could say

`ERROR: Can not find rack named python with the credentials provided (~/.aws/credentials).`

**Comment by MiguelMoll** [comment]
I really like the rewording. 👍 Might take some reworking to know if the creds came from env vars or a credentials file. Perhaps we can display **JUST** the access id? Hesitant though.

**Comment by soulshake** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [CLI Installer CSV Broken](https://trello.com/c/OvHFfNmL/1514-cli-installer-csv-broken)
